src/IECoreRenderMan/Renderer.cpp

```diff
@@ -0,0 +1,254 @@
+
+			ConstGeometryPrototypePtr geometryPrototype;
+			if( auto mesh = runTimeCast<const MeshPrimitive>( object ) )
+			{
+				// RenderMan refuses to share mesh prototypes between GeometryInstances and
+				// LightInstances, so we insert some blind data to give the mesh geometry
+				// a different hash, causing the GeometryPrototypeCache to create a prototype
+				// that won't be used by `Renderer::object()`.
+				MeshPrimitivePtr meshCopy = mesh->copy();
+				meshCopy->blindData()->writable().insert( g_forMeshLightBlindData );
+				geometryPrototype = m_geometryPrototypeCache->get( meshCopy.get(), typedAttributes, /* messageContext = */ name );
+			}
+
+			return new IECoreRenderMan::Light( geometryPrototype, typedAttributes, m_session );
+		}
+
+		ObjectInterfacePtr lightFilter( const std::string &name, const IECore::Object *object, const AttributesInterface *attributes ) override
+		{
+			const IECore::MessageHandler::Scope messageScope( m_messageHandler.get() );
+			return nullptr;
+		}
+
+		Renderer::ObjectInterfacePtr object( const std::string &name, const IECore::Object *object, const AttributesInterface *attributes ) override
+		{
+			if( !object )
+			{
+				return nullptr;
+			}
+
+			const IECore::MessageHandler::Scope messageScope( m_messageHandler.get() );
+			acquireSession();
+
+			auto typedAttributes = static_cast<const Attributes *>( attributes );
+			ConstGeometryPrototypePtr geometryPrototype = m_geometryPrototypeCache->get( object, typedAttributes, /* messageContext = */ name );
+			if( !geometryPrototype )
+			{
+				return nullptr;
+			}
+
+			return new IECoreRenderMan::Object( geometryPrototype, typedAttributes, m_session );
+		}
+
+		ObjectInterfacePtr object( const std::string &name, const std::vector<const IECore::Object *> &samples, const std::vector<float> &times, const AttributesInterface *attributes ) override
+		{
+			const IECore::MessageHandler::Scope messageScope( m_messageHandler.get() );
+			acquireSession();
+
+			auto typedAttributes = static_cast<const Attributes *>( attributes );
+			ConstGeometryPrototypePtr geometryPrototype = m_geometryPrototypeCache->get( samples, times, typedAttributes, /* messageContext = */ name );
+			if( !geometryPrototype )
+			{
+				return nullptr;
+			}
+
+			return new IECoreRenderMan::Object( geometryPrototype, typedAttributes, m_session );
+		}
+
+		void render() override
+		{
+			const IECore::MessageHandler::Scope messageScope( m_messageHandler.get() );
+			acquireSession();
+			m_materialCache->clearUnused();
+			m_globals->render();
+		}
+
+		void pause() override
+		{
+			const IECore::MessageHandler::Scope messageScope( m_messageHandler.get() );
+			m_globals->pause();
+		}
+
+		IECore::DataPtr command( const IECore::InternedString name, const IECore::CompoundDataMap &parameters ) override
+		{
+			const IECore::MessageHandler::Scope messageScope( m_messageHandler.get() );
+			return nullptr;
```

Comment:
Even though we don't handle any commands here, is it worth warning about unknown commands like we do for the other renderer backends?

Reply:
Good idea - done in https://github.com/GafferHQ/gaffer/pull/6271/commits/c03de0b16f0894570ebe73bcfbfa402e7f58d5aa.